src/ripper.rb

```diff
@@ -377,6 +377,11 @@ module Strings
     def on_tstring_end(quote)
       last_sexp.merge!(quote: quote)
     end
+
+    def on_label_end(quote)
+      # `quote` is ": or ':
+      last_sexp.merge!(quote: quote[0])
```

Comment:
i'll be honest - i have no idea where `on_tstring_end` strips the colon - when I check the value of `quote` in the method, it also has a `:` on it, but in the parser output, it's gone.